src/univalue/univalue_read.cpp

```diff
@@ -377,9 +377,11 @@ bool UniValue::read(const char *raw)
         default:
             return false;
         }
-    }
+    } while (!stack.empty ());
 
-    if (stack.size() != 0)
+    /* Check that nothing follows the initial construct (parsed above).  */
+    tok = getJsonToken(tokenVal, consumed, raw);
```

Comment:
Nit: I'd move the definitions of tokenval and consumed here, as they're only used here.